Installed Apps page revamp

leog opened the pull request:
## What does this PR do?

Revamping "Installed Apps" page to just show necessary things.

Fixes #2715
Fixes #2571

Loom Video: https://www.loom.com/share/0870dda91c484e87995a0be26c7f646f
Loom Video follow-up: https://www.loom.com/share/8531837b3eb44506bcd42b9eccdc357d

## Type of change

<!-- Please delete options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Other

## How should this be tested?

- Go to "Installed Apps" page and see only installed apps shown
- New empty state would be visible if no installed app is present
- Webhooks were moved to "Developer" tab in Settings along with API Keys (moved from "Security" tab)
- Connect additional calendars section in Installed Apps page was moved to a new dropdown next to default calendar dropdown

leog committed:
First round of changes

leog committed:
Some missing styling

leog committed:
Last round of core changes

leog committed:
Color tweaks

leog commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,77 @@
+
+type AppOutput = inferQueryOutput<"viewer.integrations">["items"][0];
+
+const AdditionalCalendarSelector = ({ isLoading }: Props): JSX.Element | null => {
```
New selector for additional connections

leog replied:
Seems like the Apple Calendar Install Button returns the dialog to be opened, which never gets a chance to be opened as the dropdown list where the install button is included (as well as the dialog component) gets closed upon option selection. Tried a bunch of ways, got very very close, but I think the code would be too hard to follow even if we can make it work. Any suggestions on how to handle the extra element returned by the Install Button? Would it be too bad of a UX redirecting the user to some place with the dialog opened as a page? Or perhaps we can make it so the dialog becomes some kind of tooltip in this dropdown case 🤔 @Jaibles your input would be highly appreciated!

zomars replied:
For now it would be better to redirect to `/apps/apple-calendar` than doing nothing at all...

leog replied:
Wouldn't work, the app is already installed, so the button is disabled to be able to click it to add/connect a new Apple Calendar:
<img width="1081" alt="Cal_com___Cal_com" src="https://user-images.githubusercontent.com/467258/169672477-7d1aac91-2121-451a-b075-0ee3749488a4.png">

zomars replied:
We need to fix the dialog to merge then

leog commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,77 @@
+          type: item.type,
+        }));
+        const customStyles = {
+          // Forgive me father for I have sinned, "any" is necessary here
```
😇

alannnc replied:
I think it would be good we use this comment to explain why any it's necessary here so in the future if another one takes up this code can understand easily what its happening here

leog replied:
I actually tried to make it not `any` but I couldn't find a type to cover it. This is an object from `react-select` library to handle custom styles over their exported Select element. It's not that important, but I'm happy to include that description if we think it helps.

zomars replied:
There's no benefit extracting the variable here. Addressed on my other comment.

leog replied:
Gotcha!

leog commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,77 @@
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
+  return (
+    <InstallAppButton
```
Kudos @zomars, the Install buttons are very flexible!

leog commented on apps/web/components/DestinationCalendarSelector.tsx:
```diff
@@ -50,6 +50,21 @@ const DestinationCalendarSelector = ({
         value: `${cal.integration}:${cal.externalId}`,
       })),
     })) ?? [];
+  const customStyles = {
```
Taking the opportunity to fix colors in the destination calendar selector as well

leog commented on apps/web/components/integrations/CalendarListContainer.tsx:
```diff
@@ -207,39 +210,44 @@ export function CalendarListContainer(props: { heading?: false }) {
-      )}
-      <ConnectedCalendarsList onChanged={onChanged} />
-      {!!query.data?.connectedCalendars.length && (
-        <ShellSubHeading
```
This is what was moved to the new dropdown

leog commented on apps/web/components/webhook/WebhookListContainer.tsx:
```diff
@@ -23,45 +22,33 @@ export type WebhookListContainerType = {
       query={query}
       success={({ data }) => (
-        <>
-          <ShellSubHeading className="mt-10" title={props.title} subtitle={props.subtitle} />
```
Simplifying the UI here to comply with figma designs

leog commented on apps/web/components/webhook/WebhookListItem.tsx:
```diff
@@ -41,8 +41,8 @@ export default function WebhookListItem(props: { webhook: TWebhook; onEditWebhoo
-                    "w-max rounded-sm px-1 text-xs ",
-                    props.webhook.active ? "bg-blue-100 text-blue-700" : "bg-blue-50 text-blue-200"
+                    "w-max rounded-sm px-1 text-xs",
+                    props.webhook.active ? "text-grey-200 bg-gray-200" : "bg-grey-50 text-grey-200"
```
Asked Ciarán if this blue was intended and he suggested to go for gray, so I changed it

hariombalhara replied:
```suggestion
                    props.webhook.active ? "text-gray-200 bg-gray-200" : "bg-gray-50 text-gray-200"
```

leog commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -81,122 +81,92 @@ function ConnectOrDisconnectIntegrationButton(props: {
+const IntegrationsContainer = ({ variant, className = "" }: IntegrationsContainerProps): JSX.Element => {
   const { t } = useLocale();
-  const query = trpc.useQuery(["viewer.integrations"], { suspense: true });
+  const query = trpc.useQuery(["viewer.integrations", { variant }], { suspense: true });
```
As mentioned, we only get the integrations per variant

zomars replied:
Can we add an input to the query it only returns installed apps? Instead of filtering on the frontend, let's just reduce the data returned from the server.

leog replied:
Makes sense!

leog commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -81,122 +81,92 @@ function ConnectOrDisconnectIntegrationButton(props: {
   const { t } = useLocale();
-  const query = trpc.useQuery(["viewer.integrations"], { suspense: true });
+  const query = trpc.useQuery(["viewer.integrations", { variant }], { suspense: true });
+  const installedFilter = (app: AppOutput) => app.credentialIds.length > 0 || app.isGlobal;
```
And we need to make sure we filter out not installed apps, as the Installed Apps page was repurposed to only show that

leog committed:
Improving code readability

leog commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -250,21 +220,46 @@ function Web3ConnectBtn() {
+                    <Web3Container />
+                  </>
+                ) : (
+                  <EmptyScreen
```
New empty state

leog commented on apps/web/server/routers/viewer.tsx:
```diff
@@ -620,27 +624,11 @@ const loggedInViewerRouter = createProtectedRouter()
-      const payment = apps.flatMap((item) => (item.variant === "payment" ? [item] : []));
-      const other = apps.flatMap((item) => (item.variant.startsWith("other") ? [item] : []));
-      const calendar = apps.flatMap((item) => (item.variant === "calendar" ? [item] : []));
+      const flapApps = apps.flatMap((item) => (item.variant === variant ? [item] : []));
```
Changes to only return the requested variant

leog committed:
Reverting unneeded changes

leog committed:
Reverting yarn.lock

leog committed:
Removing yarn.lock

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Empty state updated

leog committed:
Fixing webhook test

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Cleaning up code

leog committed:
Merge branch 'leog/installed-apps-revamp' of github.com-leog:calcom/cal.com into leog/installed-apps-revamp

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Fixing test and simplifying code a bit

leog committed:
Merge branch 'leog/installed-apps-revamp' of github.com-leog:calcom/cal.com into leog/installed-apps-revamp

leog committed:
Merging API Keys into developer section

leog committed:
Unifying Empty Screen with monorepo version

leog committed:
Cleaning up

leog commented on apps/web/pages/availability/index.tsx:
```diff
@@ -2,12 +2,12 @@ import { ClockIcon } from "@heroicons/react/outline";
 
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import showToast from "@calcom/lib/notification";
+import EmptyScreen from "@calcom/ui/EmptyScreen";
```
Took the opportunity to completely move EmptyScreen to monorepo

leog commented on apps/web/pages/settings/developer.tsx:
```diff
@@ -0,0 +1,19 @@
+    <Shell heading={t("developer")} subtitle={t("manage_developer_settings")}>
+      <SettingsShell>
+        <WebhookListContainer title={t("webhooks")} subtitle={t("receive_cal_meeting_data")} />
+        <ApiKeyListContainer />
```
As mentioned in the chat, and approved by Ciarán, API Keys live in Developers tab in Settings now

leog commented on apps/web/playwright/fixtures/users.ts:
```diff
@@ -145,12 +145,12 @@ export async function login(
 
 export async function getPaymentCredential(page: Page) {
-  await page.goto("/apps/installed");
+  await page.goto("/apps/stripe");
```
You can't install anything from "Installed Apps" anymore, making it so it goes directly to Stripe app page

leog commented on apps/web/playwright/integrations.test.ts:
```diff
@@ -24,7 +24,7 @@ test.describe("Integrations", () => {
     const [eventType] = user.eventTypes;
     await user.login();
-    await page.goto("/apps/installed");
+    await page.goto("/settings/developer");
```
Webhook is under Developer tab in Settings now

leog commented on apps/web/public/static/locales/en/common.json:
```diff
@@ -771,7 +776,6 @@
   "api_keys": "API Keys",
   "api_key_modal_subtitle": "API keys allow you to make API calls for your own account.",
   "api_keys_subtitle": "Generate API keys to use for accessing your own account.",
-  "generate_new_api_key": "Generate new API key",
```
To be consistent with Webhooks section in Developer tab in Settings, I removed the text along side the "plus" icon

leog committed:
Installed apps logic consistency + cleaning up

leog commented:
Making the loom video I noticed that now that integration apps get fetched separately due to changes I introduced, the loading experience is a bit weird, maybe I can go back to what was before, thoughts?

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

PeerRich reviewed (approved):
looks good, do you have a screenshot of the empty page?

PeerRich commented:
you can still add multiple google calendars right?

leog commented:
> you can still add multiple google calendars right?

Of course!

zomars commented:
Does this make Daily video opt-in as well? Do we need to install it from the App Store to use it as a location? Isn't the Expected behavior to be installed by default?

leog commented:
> Does this make Daily video opt-in as well? Do we need to install it from the App Store to use it as a location? Isn't the Expected behavior to be installed by default?

All very good questions. This doesn't change the status for Daily video. An app is considered installed when credentials exist in the database; we could use this PR to, for once and for all, clear things up regarding the concepts of installing, connecting, adding and the `installed` flag which I think I saw some deprecation notice somewhere. Either way, additional effort will be required to achieve consistency for the Daily video app, I'm up for it! @PeerRich @Jaibles please advice.

alannnc commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -96,122 +95,96 @@ function ConnectOrDisconnectIntegrationButton(props: {
   const { t } = useLocale();
-
+  const result = trpc.useQuery(["viewer.web3Integration"]);
+  const isWeb3Active = result.data?.isWeb3Active as boolean;
```
```suggestion
  const isWeb3Active = !!result.data?.isWeb3Active;
```
If you want to convert it to strick boolean you can use double negation

zomars replied:
What @alannnc said

Jaibles commented:
We can have Daily pre installed. Overall the goal was not to start with a blank canvas here, but just to not show things that are not installed so we can pre-install Daily.

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

zomars commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,80 @@
+            name={"additionalCalendar"}
+            placeholder={t("connect_additional_calendar")}
+            options={options}
+            styles={customStyles}
```
Just inline it. We're not receiving benefit for extracting this to a variable and losing type inference.

```suggestion
            styles={{
              placeholder: (defaultStyles) => {
                return {
                  ...defaultStyles,
                  color: "#3E3E3E",
                  marginLeft: "3px",
                };
              },
            }}
```

zomars commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,80 @@
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
```
Try using the generic instead of casting first:

```suggestion
const ImageOption = (props: OptionProps<{ [key: string]: string; type: App["type"] }>) => {
  const { data } = props;
```

zomars committed:
Type fixes

zomars commented on apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:
```diff
@@ -23,21 +24,24 @@ export default function ApiKeyListContainer() {
           {data.length > 0 && (
-            <List className="pb-6">
+            <List className="mt-6">
               {data.map((item: any) => (
```
Pushed a fix for this any

zomars commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -96,122 +95,96 @@ function ConnectOrDisconnectIntegrationButton(props: {
+            )}
+          </>
+        );
+      }}></QueryCell>
```
```suggestion
      }}/>
```

zomars commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -265,21 +238,45 @@ function Web3ConnectBtn() {
+            </AppsShell>
+          </Shell>
+        );
+      }}></QueryCell>
```
```suggestion
      }}/>
```

zomars commented on apps/web/server/routers/viewer.tsx:
```diff
@@ -620,27 +624,13 @@ const loggedInViewerRouter = createProtectedRouter()
-      const payment = apps.flatMap((item) => (item.variant === "payment" ? [item] : []));
-      const other = apps.flatMap((item) => (item.variant.startsWith("other") ? [item] : []));
-      const calendar = apps.flatMap((item) => (item.variant === "calendar" ? [item] : []));
+      let flatApps = apps;
```
You can just add a `let` to `const apps` above.

zomars reviewed (changes requested):
Impressive work @leog Let's just address some quality improvements first. 🙏🏽

leog committed:
Apply suggestions from code review

Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: alannnc <alannnc@gmail.com>

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog commented on apps/web/components/webhook/WebhookListContainer.tsx:
```diff
@@ -21,47 +19,34 @@ export type WebhookListContainerType = {
-  const { t } = useLocale();
   const query = props.eventTypeId
-    ? trpc.useQuery(["viewer.webhook.list", { eventTypeId: props.eventTypeId }], {
-        suspense: true,
```
Sorry, removed by accident

leog commented on apps/web/components/DestinationCalendarSelector.tsx:
```diff
@@ -50,6 +50,22 @@ const DestinationCalendarSelector = ({
       })),
     })) ?? [];
+  const customStyles = {
+    // Forgive me father for I have sinned, "any" is necessary here
```
Yep, totally

leog committed:
Improvements, still WIP

zomars commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,70 @@
+            placeholder={t("connect_additional_calendar")}
+            options={options}
+            styles={{
+              placeholder: (defaultStyles: any) => ({
```
```suggestion
              placeholder: (defaultStyles) => ({
```

zomars commented on apps/web/components/AdditionalCalendarSelector.tsx:
```diff
@@ -0,0 +1,70 @@
+import { Button } from "@calcom/ui";
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
```
```suggestion
import { trpc } from "@lib/trpc";
```

zomars commented on apps/web/components/DestinationCalendarSelector.tsx:
```diff
@@ -67,6 +67,18 @@ const DestinationCalendarSelector = ({
         placeholder={!hidePlaceholder ? `${t("select_destination_calendar")}:` : undefined}
         options={options}
+        styles={{
+          option: (defaultStyles: any, state: any) => ({
```
```suggestion
          option: (defaultStyles, state) => ({
```

zomars commented on apps/web/components/integrations/CalendarListContainer.tsx:
```diff
@@ -1,19 +1,20 @@
 import { QueryCell } from "@lib/QueryCell";
-import { useLocale } from "@lib/hooks/useLocale";
-import { trpc } from "@lib/trpc";
+import { trpc, inferQueryOutput } from "@lib/trpc";
```
```suggestion
import { trpc } from "@lib/trpc";
```

zomars commented on apps/web/pages/apps/installed.tsx:
```diff
@@ -96,122 +95,93 @@ function ConnectOrDisconnectIntegrationButton(props: {
+                  }
+                />
+                <List>
+                  {data.items.map((item: AppOutput) => (
```
```suggestion
                  {data.items.map((item) => (
```

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Update apps/web/pages/apps/installed.tsx

Co-authored-by: Omar López <zomars@me.com>

leog committed:
Apply suggestions from code review

Co-authored-by: Omar López <zomars@me.com>

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
App active install status

leog committed:
Apple Calendar setup, Daily.co preinstalled

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Final pass

leog committed:
Merge branch 'leog/installed-apps-revamp' of github.com:calcom/cal.com into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Minor tweaks

zomars commented on packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql:
```diff
@@ -0,0 +1 @@
+insert into "App" ("slug", "dirName", "updatedAt", "categories") values ('daily-video', 'dailyvideo', NOW(), '{video}')
```
This migration is breaking builds. Removed `autoupdate`. Let's abstain to push until this is resolved.

leog committed:
Conflicts with migration ignored

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Fixing merge

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Fixing merge yet again

leog commented:
Follow-up Loom video for the added changes in scope: https://www.loom.com/share/8531837b3eb44506bcd42b9eccdc357d

leog commented on apps/web/components/DestinationCalendarSelector.tsx:
```diff
@@ -52,23 +68,29 @@ const DestinationCalendarSelector = ({
     })) ?? [];
   return (
     <div className="relative" title={`${t("select_destination_calendar")}: ${selectedOption?.label || ""}`}>
-      {/* There's no easy way to customize the displayed value for a Select, so we fake it. */}
```
Simplifying this part which was kind of hacky with a built-in way to handle it

leog commented on apps/web/components/integrations/CalendarListContainer.tsx:
```diff
@@ -207,39 +174,54 @@ export function CalendarListContainer(props: { heading?: false }) {
-    </>
+    <QueryCell
+      query={query}
+      customLoader={<SkeletonLoader className="mt-10" />}
```
Adding missing skeleton loader for better UX

leog commented on apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
-                  ))}
-                </List>
-              )}
+    <QueryCell
```
Also went for using standard Shells to handle this type of sections for consistency

leog commented on apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
+      customLoader={<SkeletonLoader />}
+      success={({ data }) => (
+        <div className="border-b border-gray-200 py-8 pl-2 pr-1">
+          <ShellSubHeading
```
Also applied the standard Shells we already have to handle this type of sections for consistency

leog commented on apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
-          </Button>
-        </div>
-      </div>
-      <LicenseRequired>
```
Bad merge, fixing this!

leog committed:
Adopting main changes

leog commented on packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql:
```diff
@@ -0,0 +1,3 @@
+insert into "App" ("slug", "dirName", "updatedAt", "categories") 
+values ('daily-video', 'dailyvideo', CURRENT_TIMESTAMP, '{video}') 
+on conflict do nothing
```
This covers having Cal Video installed by default. Follow-up changes will tackle removing globally installed apps to also be able to uninstall them, no special treatment.

leog commented on packages/ui/skeleton/index.tsx:
```diff
@@ -27,9 +27,17 @@ const SkeletonText: React.FC<SkeletonBaseProps> = ({ width, height }) => {
+    </SkeletonContainer>
+  );
+};
+
```
Needed a pulsing button for the loading Install button in App pages

leog commented on packages/app-store/applecalendar/pages/setup/index.tsx:
```diff
@@ -0,0 +1,100 @@
+  return (
+    <div className="flex h-screen bg-gray-200">
+      <div className="m-auto rounded bg-white p-5 md:w-[560px] md:p-10">
+        <div className="flex flex-col space-y-5 md:flex-row md:space-y-0 md:space-x-5">
```
New way to handle setup for apps, thanks for pointing to it @zomars. Made it responsive.

leog committed:
Removing unneeded data-testid

hariombalhara commented:
Bug: Full page reloads when switching to installed apps for the first time
https://www.loom.com/share/08a0e1543184491daa169f1a96819fb4

hariombalhara commented on apps/web/components/apps/SkeletonLoader.tsx:
```diff
@@ -4,10 +4,10 @@ import { SkeletonText } from "@calcom/ui";
   return (
     <>
-      <ShellSubHeading title={<div className="h-6 w-32 rounded-sm bg-gray-100"></div>} />
+      <ShellSubHeading title={<div className="h-6 w-32 rounded-sm bg-gray-100"></div>} {...{ className }} />
```
Interesting way to add a property

leog replied:
haha I was kind of lazy it seems, and instead of duplicating the name of the property and its value, it came to me to do this

zomars replied:
It's ok to do this, but let's not fall into common pitfalls when doing this. I've seen how ugly this rabbit hole can go when abused.

hariombalhara reviewed (changes requested):
Massive PR !! Awesome work @leog !!

Left a few comments, bugs are all together in this comment.
https://github.com/calcom/cal.com/pull/2751#issuecomment-1140969121

Feel free to address which are important.

leog committed:
Fixing reported bugs

zomars commented on apps/web/components/webhook/WebhookListContainer.tsx:
```diff
@@ -21,47 +20,35 @@ export type WebhookListContainerType = {
-        suspense: true,
-      });
-
+    ? trpc.useQuery(["viewer.webhook.list", { eventTypeId: props.eventTypeId }])
```
Agreed with @hariombalhara

leog replied:
Happy to help even if it's not code introduced by me hehe. Simplified.

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog commented:
> **Bug-1**: Full page reloads when switching to installed apps for the first time

Fixed.

> **Bug-2**: Google Meet doesn't show up in Installed Apps when in fact I think it is installed at a global level and can't be uninstalled.

Fixed, marked Google Video (Meet) as global for now.

> **Bug-3**(On Production as well):
> The actual count of Other(6 and and Calendar apps is less than that of shown on App Store main page.

Doesn't happen for me, locally nor in production 🤔

> **Bug-4**: Disabling webhook doesn't gray out the entire tile

Fixed.

> **Suggestion-1**: How about adding a loader here. When the page is loading it feels like there is nothing here to load and it is broken.

Actually, I put a skeleton button there, pulsing to let the user know it will come up eventually. Maybe it's not enough?

https://user-images.githubusercontent.com/467258/171023589-0487c51d-0183-421c-84d4-08106baacf31.mov

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Simplifying webhook query

leog committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Merge branch 'leog/installed-apps-revamp' of github.com-leog:calcom/cal.com into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

PeerRich committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

Jaibles commented:
> Suggestion-2: Teams seem more important to me to come just next to Profile. Considering what you would use more frequently should come to the left side, it should not be after Developer or Security section. cc @Jaibles thoughts ?

Regarding this... happy for Teams to be the first thing after Profile.

kodiakhq[bot] committed:
Merge branch 'main' into leog/installed-apps-revamp

leog committed:
Moving teams settings tab to second
